src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,374 @@
+
+    private enum DistroType
+    {
+        Rpm,
+        Deb
+    }
+
+    private ITestOutputHelper OutputHelper { get; set; }
+
+    public LinuxInstallerTests(ITestOutputHelper outputHelper)
+    {
+        OutputHelper = outputHelper;
+        _dockerHelper = new DockerHelper(OutputHelper);
+
+        _tmpDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        Directory.CreateDirectory(_tmpDir);
+        _contextDir = Path.Combine(_tmpDir, Path.GetRandomFileName());
+        Directory.CreateDirectory(_contextDir);
+
+        InitializeContext();
+    }
+
+    public void Dispose()
+    {
+        try
+        {
+            Directory.Delete(_tmpDir, recursive: true);
+        }
+        catch
+        {
+        }
+    }
+
+    [Fact]
+    public void RunScenarioTestsForAllDistros()
+    {
+        if (Config.TestRpmPackages)
+        {
+            TestAllDistros(DistroType.Rpm);
+        }
+
+
+        if (Config.TestDebPackages)
+        {
+            TestAllDistros(DistroType.Deb);
+        }
+    }
+
+    private void TestAllDistros(DistroType distroType)
+    {
+        foreach (string image in (distroType == DistroType.Rpm ? RpmDistroImages : DebDistroImages))
+        {
+            try
+            {
+                OutputHelper.WriteLine($"Begin testing distro: {image}");
+                DistroTest(image, distroType);
+                OutputHelper.WriteLine($"Finished testing distro: {image}");
+            }
+            catch (Exception ex)
+            {
+                Assert.Fail($"Test failed for {image}: {ex}");
+            }
+        }
+    }
+
+    private void InitializeContext()
+    {
+        // For rpm enumerate RPM packages, excluding those that contain ".cm." in the name
+        List<string> rpmPackages = Directory.GetFiles(Config.AssetsDirectory, "*.rpm", SearchOption.AllDirectories)
+            .Where(p => !Path.GetFileName(p).Contains("-cm.") && !Path.GetFileName(p).EndsWith("azl.rpm"))
+            .ToList();
+
+        foreach (string rpmPackage in rpmPackages)
+        {
+            File.Copy(rpmPackage, Path.Combine(_contextDir, Path.GetFileName(rpmPackage)));
+        }
+
+        // Copy all DEB packages as well
+        foreach (string debPackage in Directory.GetFiles(Config.AssetsDirectory, "*.deb", SearchOption.AllDirectories))
+        {
+            File.Copy(debPackage, Path.Combine(_contextDir, Path.GetFileName(debPackage)));
+        }
+
+        // Download NetStandard 2.1 packages
+        DownloadFileAsync(NetStandard21RpmPackage, Path.Combine(_contextDir, Path.GetFileName(NetStandard21RpmPackage))).Wait();
+        DownloadFileAsync(NetStandard21DebPackage, Path.Combine(_contextDir, Path.GetFileName(NetStandard21DebPackage))).Wait();
+
+        // Copy nuget packages
+        string nugetPackagesDir = Path.Combine(_contextDir, "packages");
```

Comment:
Isn't this a sizeable copy?  Could a docker mount be utilized instead?  It would be more efficient.

Reply:
Good idea. Yes this would copy more than a GB of data.

Reply:
This copy is actually required, as we need to produce a flat layout of packages, for nuget restore use. Provided packages are likely in a tree-level structure, as is the case with `packages` dir in the `artifacts` dir.

It would be possible to keep the original directory structure and introduce many local sources in nuget.config, one source for each of the sub-directories that contain *.nuget packages. It would make code more complex, with limited performance benefits, about 1 second per test.

Reply:
I did try this approach and it worked when sharing a directory from my Windows host machine, but it failed in container build when sharing files on separate docker volume. This does complicate things quite a bit, as test container needs to mount all volumes that build container has.

I think we should proceed without this small improvement as it is not worth this extra effort.